src/services/dexes/uniswapV3/UniswapV3MultiFetcher.ts

```diff
@@ -0,0 +1,148 @@
+import {StaticJsonRpcProvider} from '@ethersproject/providers';
+import {inject, injectable} from 'inversify';
+import {Logger} from 'winston';
+import {BigNumber} from 'ethers';
+import {readFileSync} from 'fs';
+import {fileURLToPath} from 'url';
+import path from 'path';
+
+import {DexProtocolName} from '../../../types/Dexes.js';
+import {ChainsIds} from '../../../types/ChainsIds.js';
+import {UniswapV3PoolRepository} from '../../../repositories/UniswapV3PoolRepository.js';
+import Settings from '../../../types/Settings.js';
+import {ContractAddressService} from '../../../services/ContractAddressService.js';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+export type UniswapV3MultiFetcherParams = {
+  fromChain: string[];
+  base: string;
+  quote: string;
+  amountInDecimals: number;
+};
+
+type UniswapV3ContractHelperInput = {
+  pool: string;
+  base: string;
+  quote: string;
+  amountInDecimals: number;
+};
+export interface OutputValues {
+  base: string;
+  quote: string;
+  value: number;
+}
+
+@injectable()
+class UniswapV3MultiFetcher {
+  provider: StaticJsonRpcProvider | undefined;
+  readonly dexProtocol = DexProtocolName.UNISWAP_V3;
+
+  @inject('Logger') protected logger!: Logger;
+  @inject('Settings') protected settings!: Settings;
+  @inject(UniswapV3PoolRepository) protected uniswapV3PoolRepository!: UniswapV3PoolRepository;
+  @inject(ContractAddressService) contractAddressService!: ContractAddressService;
+
+  async apply(inputs: UniswapV3MultiFetcherParams[]): Promise<OutputValues[]> {
+    this.logger.debug(`[UniswapV3MultiFetcher]: start with inputs ${JSON.stringify(inputs)}`);
+
+    const poolsToFetch = await this.getPoolsToFetch(inputs);
+
+    if (poolsToFetch.size === 0) {
+      this.logger.error('[UniswapV3MultiFetcher] no data to fetch');
+      return [];
+    }
+
+    const prices = await Promise.all(
+      [...poolsToFetch.entries()].map(([chainId, pools]) => this.fetchData(chainId, pools)),
+    );
+
+    return prices.flat();
+  }
+
+  public async getPoolsToFetch(
```

Comment:
we need this method to be public?

Reply:
I did in the past to be able to test it, I remember @gagiuntoli requested to be more unit test. But I'm not sure; I think it could be private, and we test only the the apply method. Let me know what you think about it

Reply:
I changed to private and removed the unit test, I keep the scenarios on apply.
I have the impression the test was duplciated at the end. Anything I can revert, just let me know

Reply:
private is fine.